app/forms/openid_connect_authorize_form.rb

```diff
@@ -157,17 +166,12 @@ def validate_verified_within_duration
     false
   end
 
+  def ial_for_identity_record
+    ial == 2 && service_provider.liveness_checking_required ? 3 : ial
+  end
+
   def ial
-    case acr_values.sort.max
-    when Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF
-      0
-    when Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF,
-        Saml::Idp::Constants::LOA1_AUTHN_CONTEXT_CLASSREF
-      1
-    when Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF,
-        Saml::Idp::Constants::LOA3_AUTHN_CONTEXT_CLASSREF
-      2
-    end
+    Saml::Idp::Constants::AUTHN_CONTEXT_CLASSREF_TO_IAL[acr_values.sort.max]
```

Comment:
should this be a `.fetch` so that it errors? Or should we add a `|| 1` or some sort of default value?

Reply:
It did not have a default previously because the acr_values are checked up front.